### DISCUSSION TITLE
[SOLVED]: Large gutter after 2024.1 upgrade

### DISCUSSION THREAD

**Discussion by luispabon** [AtomMaterialUI/material-theme-issues#489] [Q&A] [answered]
### Describe the bug

**Material UI disabled:**
![image](https://github.com/mallowigi/material-theme-issues/assets/6388823/e0f021e1-aca2-4989-8850-0219b45488c0)

**Material UI enabled:**
![image](https://github.com/mallowigi/material-theme-issues/assets/6388823/bc8b737f-b872-4989-a905-8db0fb5d32ed)

**Observations:**
 * The partition size seems proportional to window size, resizing the window changes the panel size
 * Issue happens with and without the new UI enabled

### To Reproduce

1. Install pycharm 2024.1
2. Install material UI
3. Profit

### Expected behavior

Line numbers panel is the correct size.

### Material Theme Version

v9.1.0

### IDE

PyCharm

### System Info

Ubuntu 23.10

```shell
PyCharm 2024.1 (Professional Edition)
Build #PY-241.14494.241, built on March 28, 2024
Licensed to  [ ... ]
Runtime version: 17.0.10+8-b1207.12 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.2.0-37-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 3974M
Cores: 8
Registry:
  ide.balloon.shadow.size=0
  editor.distraction.free.mode=true
  debugger.new.tool.window.layout=true
  ide.new.editor.tabs.vertical.borders=true
  ide.linux.hide.native.title=false
  ide.experimental.ui=true
  ide.editor.tab.selection.animation=true
Non-Bundled Plugins:
  io.github.fstaudt.helm (0.8.2)
  net.sjrx.intellij.plugins.systemdunitfiles (223.240127.136)
  com.mnr.java.intellij.idea.plugin.base64helper (3.0.0)
  ru.sadv1r.ansible-vault-editor-idea-plugin (1.21)
  org.intellij.plugins.hcl (241.14494.150)
  de.netnexus.camelcaseplugin (3.0.12)
  lermitage.intellij.ilovedevtoys (1.10.2)
  org.jetbrains.plugins.go-template (241.14494.150)
  codes.rudolph.idea.cfg (0.3.1)
  com.intellij.ideolog (222.3.2.0)
  name.kropp.intellij.makefile (241.14494.150)
  ir.msdehghan.plugins.ansible (0.92)
  dev.meanmail.plugin.nginx-intellij-plugin (2022.1.1)
  indent-rainbow.indent-rainbow (2.2.0)
  com.intellij.kubernetes (241.14494.150)
  net.seesharpsoft.intellij.plugins.csv (3.3.0-241)
  com.intellij.ml.llm (241.14494.240)
  izhangzhihao.rainbow.brackets (2024.2.2-241)
  net.ashald.envfile (3.4.2)
  ru.adelf.idea.dotenv (2024.1)
  com.chrisrm.idea.MaterialThemeUI (9.1.0)
  com.fapiko.jetbrains.plugins.better_direnv (1.2.2)
  com.mallowigi (95.0.0)
  mobi.hsz.idea.gitignore (4.5.2)
  com.github.blarc.gitlab-template-lint-plugin (1.13.2)
Current Desktop: sway
```

**Comment by mallowigi** [comment, marked as answer]
Settings > Centered Editor

**Comment by luispabon** [comment]
That did the trick 👍🏽  I don't recall enabling that setting at any point though, perhaps a bad default for this version of intellij?

**Comment by mallowigi** [comment]
No. It's a new feature of the Material Theme UI. It's enabled by default for all users.

I need to write an article about this but bottomline, this centers the editor rather than pin it to the left. This is especially great when the tool windows are collapsed, mimicking the behavior of websites such as Reddit, Medium and so on.

I encourage you to give it a shot, alongside enabling the "Undock Tool Windows" action.

**Comment by luispabon** [comment]
The problem with it is that it eats nearly half of the viewport for that particular file to display what's basically empty space

**Comment by mallowigi** [comment]
Well, so is with the feature disabled. You get all the empty space on the right, rather than half the empty space on both sides.

Check how it looks like with the tool windows collapsed:

![CleanShot 2024-04-04 at 16 04 40@2x](https://github.com/mallowigi/material-theme-issues/assets/5015756/12f0fa76-64d6-4176-a163-b4dae7b4aefe)

**Comment by luispabon** [comment]
![image](https://github.com/mallowigi/material-theme-issues/assets/6388823/54d69a3c-86d9-4945-9da1-992cd6fe34aa)

vs
![image](https://github.com/mallowigi/material-theme-issues/assets/6388823/46cb8f44-4789-4660-8353-6d92ab41d0dc)

The former produces a predictable place for your eyes to start scanning the file, as the margin is always the same. I personally think it's not a good default and hurts ergonomy and especially for those of us already accustomed to intellij's default, but other people might disagree clearly, 100% personal pref.

**Comment by mallowigi** [comment]
It challenges the habits, that's for sure. But it's the same as when the IntelliJ new UI was introduced, a lot of people were disappointed, but they left the setting as the default for users to "catch on the trend". Obviously, I'm not pushing it without a way to revert to the old behavior.

Think of it this way: in most websites nowadays, the content is often centered in the middle. So why not while coding?

In fact, VScode already introduced that setting (though, it's not the default, I agree with that):
![CleanShot 2024-04-04 at 16 18 07@2x](https://github.com/mallowigi/material-theme-issues/assets/5015756/c4fc8b6d-0ff1-4e9d-b38c-10c2f423510c)

**Comment by AndraxDev** [comment]
> No. It's a new feature of the Material Theme UI. It's enabled by default for all users.
> 
> I need to write an article about this but bottomline, this centers the editor rather than pin it to the left. This is especially great when the tool windows are collapsed, mimicking the behavior of websites such as Reddit, Medium and so on.
> 
> I encourage you to give it a shot, alongside enabling the "Undock Tool Windows" action.

How to disable this feature? Where can I find "Undock Tool Windows" action?

**Comment by mallowigi** [comment]
> Settings > Centered Editor

**Comment by Luke-Shepp** [comment]
There's a few bug reports gone into phpstorm for this, it's also not mentioned in the plugin changelog / whats new. It's a little odd for a setting like this to be enabled by default

https://youtrack.jetbrains.com/issue/WI-76827
https://youtrack.jetbrains.com/issue/WI-76840/Line-numbers-column-width-too-wide
https://youtrack.jetbrains.com/issue/WI-76856/Incorrect-code-window-display-after-update.-34-screen


<blockquote><img src="static/favicon.ico?3c05e1a5b709165a5c4e825adf27273a?3c05e1a5b709165a5c4e825adf27273a" width="48" align="right"></blockquote>
<blockquote><img src="static/favicon.ico?3c05e1a5b709165a5c4e825adf27273a?3c05e1a5b709165a5c4e825adf27273a" width="48" align="right"></blockquote>
<blockquote><img src="static/favicon.ico?3c05e1a5b709165a5c4e825adf27273a?3c05e1a5b709165a5c4e825adf27273a" width="48" align="right"></blockquote>

**Comment by mallowigi** [comment]
Just like I said previously, it's an opinionated choice. 

I apologize for the lack of visibility in the changelog and the  documentation. I'll try to find a time to carefully document that, in the meantime, this issue will serve as a reference for future issues of the same nature.

**Comment by bariod** [comment]
I might enjoy it more if it actually were in the center of the screen, not offset to the right by the project window.

**Comment by mallowigi** [comment]
> I might enjoy it more if it actually were in the center of the screen, not offset to the right by the project window.
> 
> 

This can be achieved by executing the action "undock tool windows"

**Comment by bariod** [comment]
There isn't a way to combine the two? 
I.e. docked windows and a "truly" centered editor. This would obviously only work as long as the window is small enough to fit in the empty space to the left of the editor.

**Comment by mallowigi** [comment]
There are two separate actions. The centered editor feature does just that.

In fact, there is a similar feature like Centered Editor, available in the native IDE called "zen mode". When enabled, it undocks all tool windows and center editors. However it also hides other parts of the ide, and there is no way to control what to hide and what not. The centered editor is like in the middle, it only centers the editor. (In fact, the centered editor and zen mode use the same functionality under the hood).

**Comment by mallowigi** [comment]
Also, this behavior is consistent with the one you can find in VSCode:

![CleanShot 2024-04-05 at 10 19 18@2x](https://github.com/mallowigi/material-theme-issues/assets/5015756/1cb52184-ba10-4bbd-9a0d-ea2ea838af32)

![CleanShot 2024-04-05 at 10 19 29@2x](https://github.com/mallowigi/material-theme-issues/assets/5015756/6b0527a7-14e7-4be8-bc50-21af56268251)

**Comment by Luke-Shepp** [comment]
Definitely a useful feature on bigger monitors - but when using a 14" macbook screen I collapse all sidebars to get a full width editor, so it needs to be justified left for me. Happy that there's a toggle for it, but it was just confusing as there was nothing mentioned in the changelog when trying to find which plugin or phpstorm version was responsible 😁

**Comment by mallowigi** [comment]
Yes, I definitely apologize for the lateness in updating the documentation, there's just too much to write about for version 9.0.0 and I didn't find the right time to invest the time for it.
(it should still be available in the changelog of the plugin in the Marketplace)

**Comment by mallowigi** [comment]
> Definitely a useful feature on bigger monitors - but when using a 14" macbook screen I collapse all sidebars to get a full width editor, so it needs to be justified left for me. Happy that there's a toggle for it, but it was just confusing as there was nothing mentioned in the changelog when trying to find which plugin or phpstorm version was responsible 😁

I concur that for people who leave tool windows on the left open, they are using them as the left margin for the code to start. For such people having yet another gap might feel irrelevant (and explains somewhat why Zen Mode forces tool windows to be hidden).

Perhaps I should display a notification or a popup explaining the feature for further questions.

**Comment by luxferre** [comment]
first seen frustrating, now just nice

**Comment by mallowigi** [comment]
I've finally updated the documentation, hope this will clear up questions from now on

**Comment by juanjoseluisgarcia** [comment]
> Settings > Centered Editor

Where? how? Screenshot please

**Comment by bariod** [comment]
> > Settings > Centered Editor
> 
> Where? how? Screenshot please

`Material Theme UI > Features > Centered Editor`
https://material-theme.com/docs/what-s-new/#centered-editor

<blockquote><img src="https://www.material-theme.com/img/brand/logo.png" width="48" align="right"><div><strong><a href="https://material-theme.com/docs/what-s-new/#centered-editor">What's new? - Material Theme UI Documentation</a></strong></div><div>The Material Theme plugin is a plugin for JetBrains IDEs (IntelliJ, PhpStorm, WebStorm, Android Studio, etc...) which completely redesigns
the UI with a beautiful and configurable Material Design UI.
This is the plugin's documentation page which presents and explains the different features provided by the plugin.
</div></blockquote>

**Comment by luispabon** [comment]
> > Settings > Centered Editor
> 
> Where? how? Screenshot please

Just do a text search for it on the settings window

**Comment by lhall-amphibee** [comment]
Wow I thought my editor broken down at first. Happily reverted back the option :+1:

**Comment by Yexeed** [comment]
Yeah, this new options made me waste about ~40 minutes, I've gone through all cycles: denial, anger, trying to run without plugins, with them, disable one-by-one separatly and so on. In the end, I've decided to uninstall plugin and live forward. How lucky I was when before uninstalling this forever I've decided to search for issues page and look for new issues. My default editor appearance is that project tab is enabled in 99% of my code time, making editor "KINDA" centered all the time. Yes, I accept the fact that my habit of centering the editor has grown over the years, I just can't code undistractibly if the project tab is closed, and code goes all the way to the left of the screen. However, enabling this new option by-default when you update for such users as me, creates unmiserable confusion. You definetly should have, before enabling it silently after regular update, think it thorougly. Popping a notification is a real close solution 🤙🏼

**Comment by mallowigi** [comment]
Next version will pop-up a notification to limit confusion, I hope.

**Comment by Yexeed** [comment]
No offsense btw; None made. P.s. also when I've updated the plugin, first thing that I've noticed that this newly created blank space which centers the editor, are a very similar to those you get when clicking on 'code-author' inlay hint. That made me think that this blaming tool-window is the reason why editor has been centered, not the material theme plugin. Maybe it is possible for you to add some option to indicate about the blank space in the menu when you right-click it? Also another suggestions, file tabs remain not affected, and are staying in the left, which creates some confusion, maybe you could add an option to the settings menu to affect tabs, alongside with the editor?
![image](https://github.com/mallowigi/material-theme-issues/assets/32373441/36e70dc7-5224-4f20-8a8a-ed8d897a1112)
![image](https://github.com/mallowigi/material-theme-issues/assets/32373441/d228a9cf-e838-4096-bddf-8163730e15b0)

**Comment by mallowigi** [comment]
I'm not sure I understand what you're talking about... What is "file tabs" ?

**Comment by Yexeed** [comment]
> I'm not sure I understand what you're talking about... What is "file tabs" ?

Opened files tabs ![319579108-bc8b737f-b872-4989-a905-8db0fb5d32ed.png](https://github.com/mallowigi/material-theme-issues/assets/32373441/e9e9f9fb-f275-452e-b73d-99627b90ad8f)

**Comment by mallowigi** [comment]
So you mean an option for the tabs to be on the center as well? Unfortunately it is not possible. The toggle centers the code in the middle of the editor, but the tabs are outside the editor. Moreover I don't think this will look good if you ask me.

Let's wait the first editor to implement this and we will see if it makes sense :)

**Comment by ekvedaras** [comment]
People, just put project view on the right, no need to center anything 😃 but interesting, will try it out 🤔

**Comment by coder-wuya** [comment]
我找了一早上，终于知道是啥问题了

**Comment by XuBoZhangQvQ** [comment]
@mallowigi hi,there,I have found the root cause of the problem
In the new version, simply adjusting the settings of Material Theme Ul will open the Distraction Free Mode which should be a bug in the new version
After adjusting the settings, just manually turn off the Distraction Free Mode

<img width="810" alt="image" src="https://github.com/mallowigi/material-theme-issues/assets/35949334/4a7454a3-24bb-4180-9c70-c36537e67eb8">

**Comment by mallowigi** [comment]
> 
> 
> @mallowigi hi,there,I have found the root cause of the problem
> 
> In the new version, simply adjusting the settings of Material Theme Ul will open the Distraction Free Mode which should be a bug in the new version
> 
> After adjusting the settings, just manually turn off the Distraction Free Mode
> 
> 
> 
> <img width="810" alt="image" src="https://github.com/mallowigi/material-theme-issues/assets/35949334/4a7454a3-24bb-4180-9c70-c36537e67eb8">
> 
> 

It's not a bug, did you read the thread at all?

**Comment by Vorta** [comment]
Who in their right mind thought that this option should be on by default?

**Comment by XuBoZhangQvQ** [comment]
> > @mallowigi hi,there,I have found the root cause of the problem
> > In the new version, simply adjusting the settings of Material Theme Ul will open the Distraction Free Mode which should be a bug in the new version
> > After adjusting the settings, just manually turn off the Distraction Free Mode
> > <img alt="image" width="810" src="https://private-user-images.githubusercontent.com/35949334/320346346-4a7454a3-24bb-4180-9c70-c36537e67eb8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NzA5NDksIm5iZiI6MTcxMjU3MDY0OSwicGF0aCI6Ii8zNTk0OTMzNC8zMjAzNDYzNDYtNGE3NDU0YTMtMjRiYi00MTgwLTljNzAtYzM2NTM3ZTY3ZWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDEwMDQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3NjI2OWE3ZjE1Y2U0MGI1YzI5Y2FhNTY0N2FmZTU0YzY0Mzk4MzE2NmNmZGQyMTNhOWE4NjlkMjkxZmY4ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mLXtiKWVOFF2-MSIP3iq8wzNPa36oKsCz6v8zLLWvrY">
> 
> It's not a bug, did you read the thread at all?

sorry,I really didn't see all of this thread,But on my monitor, it does seem strange to have this feature enabled by default(4K  27 inches  3008×1692)

<img width="3008" alt="image" src="https://github.com/mallowigi/material-theme-issues/assets/35949334/5c352c37-7c91-4213-a289-3e85a8d1736a">

**Comment by mallowigi** [comment]
Changed the default to false. Gave up on challenging habits.

**Comment by mallowigi** [comment]
[Material Theme UI-9.2.0.zip](https://github.com/mallowigi/material-theme-issues/files/14904247/Material.Theme.UI-9.2.0.zip)

Is the default gone with this version?

**Comment by apb-hello-world** [comment]
I do not see the option for this under settings > features anymore. I actually like the centered editor.

**Comment by mallowigi** [comment]
I didn't remove the feature, it should still be inside the settings. Have you checked in all sections of the Features settings page?

**Comment by apb-hello-world** [comment]
> I didn't remove the feature, it should still be inside the settings. Have you checked in all sections of the Features settings page?

![image](https://github.com/mallowigi/material-theme-issues/assets/8071683/c450e5f7-975b-4e04-bd08-88b8d762cb50)

I don't see the features page where it used to be under the main Material Theme UI dropdown, only the `Features` tab in settings, and nothing in there about the centered editor.

Edit: When I open the IDE the editor is centered for a few seconds, and then moves back to the left side.
Edit2: The option is only missing for me in Rider. It is there in Webstorm and working great.

**Comment by mallowigi** [comment]
You are still using the version 8.13

**Comment by apb-hello-world** [comment]
> You are still using the version 8.13

Should Rider be showing a newer version? I see it's 9.1.0 in webstorm, 8.13.2 in Rider.

**Comment by mallowigi** [comment]
Only if you have 2024.X
